validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -89,6 +98,44 @@ def _get(self, address, create=False):
     def get(self, address):
         return self._get(address)
 
+    def walk(self, address):
+        '''
+        Returns a stream of node data, raising AddressNotInTree if
+        ADDRESS is not in the tree.
+
+        First, the ancestors of ADDRESS (including itself) are
+        yielded, earliest to latest. Next, a dummy node is yielded to
+        indicate that all the ancestors have been returned. Finally,
+        the descendants of ADDRESS are yielded in an unspecified
+        order.
```

Comment:
Instead of unspecified, can we commit to depth- or breadth-first search? Is there a reason not to?

Reply:
Is there a specific use case for one or the other? If not, why constrain the implementation?